content/_code-samples/airgapped-wallet/py/airgapped-wallet.py

```diff
@@ -0,0 +1,203 @@
+import os
+import base64
+import qrcode
+import platform
+from PIL import Image
+from pathlib import Path, PureWindowsPath, PurePath
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+from xrpl import wallet
+from xrpl.core import keypairs
+from xrpl.utils import xrp_to_drops
+from xrpl.models.transactions import Payment
+from xrpl.transaction import safe_sign_transaction
+
+
+def create_wallet():
+    """
+    Generates a keypair
+    """
+
+    seed = keypairs.generate_seed()
+    pub, priv = keypairs.derive_keypair(seed)
+
+    address = keypairs.derive_classic_address(pub)
+    private = keypairs.derive_classic_address(priv)
+    print(
+        f"\n\n       XRP WALLET CREDENTIALS"
+        f"\n  Wallet Address: {address}"
+        f"\n     Private Key: {private}"
+        f"\n            Seed: {seed}"
+          )
+
+    return address, private, seed
+
+
+def sign_transaction(_xrp_amount, _destination, _ledger_seq, _wallet_seq, password):
+    """
+    Signs transaction and returns signed transaction blob in QR code
+    """
+
+    with open(get_path("/Wallet/private.txt"), "r") as f:
+        _seed = f.read()
+        _seed = bytes.fromhex(_seed)
+
+    with open(get_path("/Wallet/salt.txt"), "rb") as f:
+        salt = f.read()
+
+    # Line 49-58: initialize key
+    kdf = PBKDF2HMAC(
+        algorithm=hashes.SHA256(),
+        length=32,
+        iterations=100000,
+        salt=salt
+    )
+
+    key = base64.urlsafe_b64encode(kdf.derive(bytes(password.encode())))
+    crypt = Fernet(key)
+    
+    # Decrypts the wallet's private key
+    _seed = crypt.decrypt(_seed)
+    
+    # Initialize XRPL wallet using decrypted private key
+    _wallet = wallet.Wallet(seed=_seed.decode(), sequence=0)
+
+    validated_seq = _ledger_seq
+    _wallet.sequence = _wallet_seq
+    
+    # Construct Payment transaction
+    my_tx_payment = Payment(
+        account=_wallet.classic_address,
+        amount=xrp_to_drops(xrp=_xrp_amount),
+        destination=_destination,
+        last_ledger_sequence=validated_seq + 100, # +100 to catch up with the ledger when we transmit the signed tx blob to Machine 2
+        sequence=_wallet.sequence,
+        fee="10"
+    )
```

Comment:
Relatedly, how does the user initially fund the wallet? (Should that be a step in the tutorial?)

Reply:
They'd have the create the wallet and fund it themselves, I'll add this step in the tutorial.

Reply:
I don't quite see this step in the tutorial yet - Marking conversation as unresolved in the meantime (I read a couple times, but if I just missed it please let me know)

Reply:
![image](https://user-images.githubusercontent.com/87929946/201845568-369d2c2b-7f79-4bcc-993a-ed45781294db.png)